Sources/Component/CollapsibleView/CollapsibleView.swift

```diff
@@ -0,0 +1,132 @@
+//  Copyright © 2020 Jamit Labs GmbH. All rights reserved.
+
+import UIKit
+
+/// Protocol a header view should conform to to be informed about certain changes
+public protocol CollapsibleHeaderViewDelegate {
+    /// Called when the state of the collapsible view has changed.
+    ///
+    ///- Parameter to: The current state of the collapsible view..
+    func didChangeCollapsibleState(to isCollapsed: Bool)
+}
+
+/// A stateful collapsible view.
+///
+/// Example:
+/// ```swift
+/// let defaultCollapsibleHeaderView: DefaultCollapsibleHeaderView = .init()
+/// defaultCollapsibleHeaderView.model = .init(
+///     title: "Header title",
+///     titleFont: .systemFont(ofSize: 16.0),
+///     arrowImageUp: nil // TODO: Place an image here if needed
+/// )
+/// // TODO: adjust height of your header view.
+/// defaultCollapsibleHeaderView.heightAnchor.constraint(equalToConstant: 60.0).isActive = true
+///
+/// let view: UIView = .init()
+/// view.backgroundColor = .red
+/// view.heightAnchor.constraint(equalToConstant: 44.0).isActive = true
+/// view.widthAnchor.constraint(equalToConstant: UIScreen.main.bounds.width).isActive = true
+///
+/// collapsibleView.model = CollapsibleViewModel(
+///     headerView: defaultCollapsibleHeaderView,
+///     items: [view],
+///     isCollapsed: true
+/// )
+/// ```
+public final class CollapsibleView: StatefulView<CollapsibleViewModel> {
+    private lazy var headerViewContainer: UIView = .init()
+
+    private lazy var actionButton: UIButton = {
+        let actionButton: UIButton = .init()
+        actionButton.setTitle("", for: .normal)
+        actionButton.addTarget(self, action: #selector(didTriggerAction), for: .primaryActionTriggered)
+        actionButton.addTarget(self, action: #selector(updateBackgroundColor), for: .allTouchEvents)
+        actionButton.addTarget(self, action: #selector(lateUpdateBackgroundColor), for: .allTouchEvents)
+        return actionButton
+    }()
+
+    private lazy var stackView: UIStackView = {
+        let stackView: UIStackView = .init()
+        stackView.alignment = .fill
+        stackView.axis = .vertical
+        stackView.distribution = .fillProportionally
+        return stackView
+    }()
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+
+        headerViewContainer.backgroundColor = .white
```

Comment:
I guess we should make it configurable, since the backgroundColor could be differ.

Reply:
Done.